app/lib/submission_builder/ty2024/states/nj/documents/nj1040.rb

```diff
@@ -137,6 +137,18 @@ def document
                     xml.TotalIncome calculated_fields.fetch(:NJ1040_LINE_27)
                   end
 
+                  if calculated_fields.fetch(:NJ1040_LINE_28A).positive? && Flipper.enabled?(:show_retirement_ui)
```

Comment:
[dust] In the spirit of data flowing from the calculator to the submission builder to the PDF, should the flipper check maybe be in the calculator instead?

Reply:
[loosely held opinion] I prefer the check being in the XML as-is over the calculator. It feels like the calculator is more "pure" logic - ie, this is what the field calculates to given the input we have. Whereas the flipper check is a UI thing - "do we show this particular information to the user in the app" hence putting it in the XML builder